adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -668,6 +670,9 @@ private Intent getIntentForBrokerActivityFromAccountManager(final Bundle addAcco
             //
             // TODO add retry logic since authenticator is not responding to
             // the request
+            // TODO (2/13/2018) -- This is kind of strange. Rather than throw an AuthenticationException here this code just returns null and lets
+            // the result-check inside of AcquireTokenWithBrokerRequest#acquireTokenWithBrokerInteractively(IWindowComponent) throw it with a
+            // different error: ADALError.DEVELOPER_ACTIVITY_IS_NOT_RESOLVED --  is this preferable? Why not just throw this Exception here?
             Logger.e(TAG + methodName, AUTHENTICATOR_CANCELS_REQUEST, "", ADALError.BROKER_AUTHENTICATOR_NOT_RESPONDING, e);
```

Comment:
@addev-ashish Take a look at the above comment and let me know what you think

Reply:
there's another TODO to add retry logic. which should already exist?

---
In reply to: [168087132](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1119#discussion_r168087132) [](ancestors = 168087132)

Reply:
to clarify, this is just a question for the AM route if we go down that path, but is not related to the Teams fix we are doing correct?

---
In reply to: [168218504](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1119#discussion_r168218504) [](ancestors = 168218504,168087132)

Reply:
Correct, this was just a question regarding an implementation quirk and doesn't affect the larger Teams changes.

Reply:
At a glance, it looks like the existing `//TODO` has not been resolved. This call does not attempt a retry currently